pegasus_wrapper/resource_request.py

```diff
@@ -146,6 +148,10 @@ def apply_to_job(self, job: Job, *, job_name: str) -> None:
                 else _DEFAULT_JOB_TIME_IN_MINUTES
             ),
         )
+
+        if exclude_list:
```

Comment:
Can you add a unit test which tests having nodes excluded in a test workflow?

Reply:
Yes.